lib/SPIRV/SPIRVToOCL.cpp

```diff
@@ -935,6 +941,60 @@ void SPIRVToOCLBase::visitCallSPIRVBFloat16Conversions(CallInst *CI, Op OC) {
   mutateCallInst(CI, Name);
 }
 
+void SPIRVToOCLBase::visitCallSPIRVDot(CallInst *CI, Op OC,
+                                       StringRef DemangledName) {
+  // OpenCL only supports integer dot product builtins that have return types
+  // of int and uint.
+  if (!(DemangledName.contains("_Rint") || DemangledName.contains("_Ruint")))
+    return;
+
+  bool IsPacked = !CI->getOperand(0)->getType()->isVectorTy();
+  std::stringstream Name;
+  switch (OC) {
+  case OpSDotKHR:
```

Comment:
nit:
```suggestion
  case OpSDot:
```
all of these KHR instructions are incorporated in SPIR-V 1.6. New instructions share the same op codes as KHR versions, so such change is NFC.